src/main/java/jenkins/plugins/publish_over/BPPlugin.java

```diff
@@ -107,7 +107,7 @@ public void perform(final Run<?, ?> build, final FilePath workspace, final Launc
         if (PROMOTION_CLASS_NAME.equals(build.getClass().getCanonicalName())) {
             AbstractBuild<?, ?> promoted;
             try {
-                final Method getTarget = build.getClass().getMethod("getTarget", (Class<?>[]) null);
+                final Method getTarget = build.getClass().getMethod("getTargetBuildOrFail", (Class<?>[]) null);
                 promoted = (AbstractBuild) getTarget.invoke(build, (Object[]) null);
```

Comment:
Something along these lines perhaps?

```suggestion
                final Method getTarget = build.getClass().getMethod("getTargetBuildOrFail", (Class<?>[])null);
                if(getTarget == null) {
                    getTarget = build.getClass().getMethod("getTarget", (Class<?>[])null);
                }
                if(getTarget != null) {
                    promoted = (AbstractBuild) getTarget.invoke(build, (Object[]) null);
                }
```

Reply:
Don't commit this suggestion, it wouldn't let me include the modified line.